File: app/src/main/java/com/cliniva/enventory/viewholder/SupplierHolder.java
package com.cliniva.enventory.viewholder;

import android.databinding.ViewDataBinding;
import android.support.annotation.NonNull;

import com.cliniva.enventory.adapter.base.BaseRecyclerViewHolder;
import com.cliniva.enventory.databinding.ItemSupplierBinding;
import com.cliniva.enventory.event.OnSupplierItemClickedListener;
import com.cliniva.enventory.model.Customer;

public class SupplierHolder extends BaseRecyclerViewHolder<Customer, OnSupplierItemClickedListener> {

    private ItemSupplierBinding binding;

    public SupplierHolder(@NonNull ViewDataBinding itemView) {
        super(itemView);
        binding = (ItemSupplierBinding) itemView;
    }

    @Override
    public void onBindView(Customer item, int position, OnSupplierItemClickedListener listener) {
        enableItemViewClick(item, listener);
        binding.setSupplier(item);
    }
}

File: app/src/main/java/com/amirsons/inventory/recyclerview/viewholder/SupplierHolder.kt
package com.amirsons.inventory.recyclerview.viewholder

import android.view.View
import com.amirsons.inventory.datamanager.model.Supplier
import com.amirsons.inventory.event.OnSupplierItemClickedListener
import com.amirsons.inventory.recyclerview.base.BaseRecyclerViewHolder
import kotlinx.android.synthetic.main.item_customer.view.tv_item_amount
import kotlinx.android.synthetic.main.item_customer.view.tv_item_invoice_date
import kotlinx.android.synthetic.main.item_supplier.view.*

class SupplierHolder(itemView: View) : BaseRecyclerViewHolder<Supplier, OnSupplierItemClickedListener>(itemView) {

    override fun onBindView(item: Supplier, position: Int, listener: OnSupplierItemClickedListener?) {

        itemView.tv_supplier_name.text = item.name
        itemView.tv_item_amount.text = item.payableAmount.toString()
        itemView.tv_item_invoice_date.text = item.lastInvoiceDate

        enableItemViewClick(item, listener)
    }
}
